android/src/main/kotlin/com/simform/flutter_credit_card/FlutterCreditCardPlugin.kt

```diff
@@ -0,0 +1,116 @@
+package com.simform.flutter_credit_card
+
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import android.hardware.SensorManager
+import android.os.Build
+import android.view.Display
+import android.view.WindowManager
+import androidx.annotation.ChecksSdkIntAtLeast
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.EventChannel
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+
+private const val GYROSCOPE_CHANNEL_NAME = "com.simform.flutter_credit_card/gyroscope"
+private const val METHOD_CHANNEL_NAME = "com.simform.flutter_credit_card"
+
+@ChecksSdkIntAtLeast(api = Build.VERSION_CODES.R)
+private val isAtLeastOsR: Boolean = Build.VERSION.SDK_INT >= Build.VERSION_CODES.R
+
+/** FlutterCreditCardPlugin */
+class FlutterCreditCardPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
+    private var activity: Activity? = null
```

Comment:
Lets move gyroscope related functionality to a separate class and register that to main activity.